aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -575,26 +415,38 @@ If a class seems to have too many injected dependencies, this is generally a sig
 
 The container calls <xref:System.IDisposable.Dispose*> for the <xref:System.IDisposable> types it creates. If an instance is added to the container by user code, it isn't disposed automatically.
 
+In the following example, the services are created by the service container and disposed automatically:
+
 ```csharp
-// Services that implement IDisposable:
 public class Service1 : IDisposable {}
 public class Service2 : IDisposable {}
-public class Service3 : IDisposable {}
 
-public interface ISomeService {}
-public class SomeServiceImplementation : ISomeService, IDisposable {}
+public interface IService3 {}
+public class Service3 : IService3, IDisposable {}
 
 public void ConfigureServices(IServiceCollection services)
 {
-    // The container creates the following instances and disposes them automatically:
     services.AddScoped<Service1>();
     services.AddSingleton<Service2>();
-    services.AddSingleton<ISomeService>(sp => new SomeServiceImplementation());
+    services.AddSingleton<IService3>(sp => new Service3());
+}
+```
+
+In the following example:
 
-    // The container doesn't create the following instances, so it doesn't dispose of
-    // the instances automatically:
-    services.AddSingleton<Service3>(new Service3());
-    services.AddSingleton(new Service3());
+* The service instances aren't created by the service container.
+* The intended service lifetimes aren't known by the framework.
+* The framework doesn't dispose of the services automatically.
+* If the services aren't explicitly disposed in developer code, they persist until the app shuts down.
```

Comment:
Is there technically any benefit to using the [Lock pattern ](https://github.com/aspnet/AspNetCore.Docs/issues/16773#issuecomment-580091374 )for explicit disposal on shutdown if possible or is the end result effectively the same without it?

Reply:
I can go with any of (based on @anurse thoughts on that question) ...

* Cross-link the Lock blog.
* Just *say* to use `ApplicationStopping` ... let's call this option 👮 *Stealing Lock!*:tm: 👮‍♀. (cc: @andrewlock)
* Do nothing.